### DISCUSSION TITLE
Deprecation of support for legacy CMake generators in Conan Center

### DISCUSSION THREAD

**Discussion by jcar87** [conan-io/conan-center-index#17406] [Announcements]
With Conan 2.0 packages now starting to be populated, it’s time to continue looking to the future. The motivation behind this is to encourage the adoption of Conan 2.0 and move towards simplifying recipe and test_package logic.

The `cmake_find_package` or `cmake_find_package_multi` generators were deprecated by the Conan client in favor of new generators `CMakeToolchain` and `CMakeDeps` - introduced with Conan 1.33. During the transition to new generators, we wanted have continued to ensure recipes were compatible with the legacy generators, in order to give users enough time to migrate. This was enabled by introducing a ``test_v1_package/`` folder in recipes in `conan-center-index` to ensure legacy generators were also tested on the consumer side.

With Conan 2.0 being released, `CMakeToolchain` and `CMakeDeps` are now stable, and they are also available for use in Conan 1.x. Today we are announcing that we will drop the need to test these generators in pull requests in Conan Center, by no longer enforcing the presence of a `test_v1_package`. This will take effect on 15 May 2023.

> **Note**
> Starting 15 May 2023, `test_v1_package` will no longer be mandatory for new recipes, nor will it be enforced moving forward.

### Key takeaways for recipe users
_New_ recipes will not be required to have logic in their `package_info()` to support legacy CMake generators, nor will they be required to have a `test_v1_package` folder.

For existing recipes, you should expected the deprecated generators to gradually lose support over time, as after a certain date they will no longer be tested by our CI service.

Thus, in order to avoid disruption, we strongly advise to migrate to the new CMake generators, as well as follow the advice in [ConanCenters “Consuming Recipes” Section](https://github.com/conan-io/conan-center-index/blob/master/docs/consuming_recipes.md) in any case.

### For recipe contributors

> **Warning**
>  Pull requests for **_existing_** recipes **will not be accepted** if they remove the `test_v1_package` folder of if they modify the `package_info()` method in such a way that it breaks support for consumers using legacy ``cmake_find_package[_multi]`` 
 generators.

* Starting on 15 May 2023, pull requests for __new recipes__ can omit a `test_v1_package`, and are not required to add anything in the `package_info()` method to support legacy generators.
* For __existing recipes__, pull requests will not be accepted if they remove or modify ``cpp_info.names`` or ``cpp_info.filenames`` from `package_info()` or actively remove `test_v1_package` in a way that it breaks support for these legacy generators. We do not intend to break any existing consumer using legacy generators, to avoid disruption.
* At a later date to be announced, all `test_v1_package` folders from `conan-center-index` will be removed and will stop being tested. This will be done by the Conan Center maintainers with no action required from our contributors.

**Comment by prince-chrismc** [comment]
This is information is still relevant but an updated announcement can be found https://github.com/conan-io/conan-center-index/discussions/19104 if there are any relevant questions :)
